components/ShoppingItem.js

```diff
@@ -1,24 +1,57 @@
 import styled from "styled-components";
 import Link from "next/link";
+import { useState } from "react";
+import Delete from "@/public/icons/delete.svg";
+
+export default function ShoppingItem({
+  id,
+  quantity,
+  name,
+  category,
+  onDeleteItem,
+}) {
+  const [isDeleteConfirmVisible, setIsDeleteConfirmVisible] = useState(false); // Zustand für den Bestätigungsdialog
+
+  // Funktion zum Anzeigen des Bestätigungsdialogs
+  function handleShowConfirm() {
+    setIsDeleteConfirmVisible(true);
+  }
+
+  // Funktion zum Verstecken des Bestätigungsdialogs
+  function handleHideConfirm() {
+    setIsDeleteConfirmVisible(false);
+  }
```

Comment:
I would combine these two functions into a toggle function. If the delete dialog is currently displayed, it should no longer be displayed; if it is not currently displayed, it should be displayed. So the opposite of the current state should always happen. You can do this with the NOT operator, e.g. like this:

```suggestion
  function handleToggleDelete() {
    setIsDeleteConfirmVisible(!isDeleteConfirmVisible);
  }
```

You could also think about the name. confirm doesn't explain that it's about the delete dialog. I would probably choose handleToggleDelete and something with show as the state, e.g. showDeleteDialog, setShowDeleteDialog or something like that?